src/soundbase.cpp

```diff
@@ -166,24 +166,36 @@ QString CSoundBase::SetDev ( const QString strDevName )
         if ( !vsErrorList.isEmpty() )
         {
             // create error message with all details
-            QString sErrorMessage = "<b>" + QString ( tr ( "No usable %1 audio device found." ) ).arg ( strSystemDriverTechniqueName ) +
-                                    "</b><br><br>" + tr ( "These are all the available drivers with error messages:" ) + "<ul>";
+            QString sErrorMessage = tr ( "<b>Could not find compatible sound driver (using %1).</b><br><br>" )
+                                          .arg ( strSystemDriverTechniqueName );
 
             for ( int i = 0; i < lNumDevs; i++ )
             {
-                sErrorMessage += "<li><b>" + GetDeviceName ( i ) + "</b>: " + vsErrorList[i] + "</li>";
+                sErrorMessage += "<b>" + GetDeviceName ( i ) + "</b>: " + vsErrorList[i] + "<br><br>";
             }
-            sErrorMessage += "</ul>";
+
 
 #ifdef _WIN32
             // to be able to access the ASIO driver setup for changing, e.g., the sample rate, we
             // offer the user under Windows that we open the driver setups of all registered
             // ASIO drivers
-            sErrorMessage += "<br/>" + tr ( "Do you want to open the ASIO driver setup to try changing your configuration to a working state?" );
+            sErrorMessage += "<br>" + tr ( "You may be able to fix errors in the driver settings." );
+            // for Windows use a QMessageBox to show the error
+            QMessageBox qmASIOWarningBox;
+            QPushButton *btnASIOSettings = qmASIOWarningBox.addButton ( tr ( "Open ASIO settings" ), QMessageBox::AcceptRole );
+
+            qmASIOWarningBox.addButton ( QMessageBox::Cancel );
+            qmASIOWarningBox.setText ( sErrorMessage );
+            qmASIOWarningBox.setIcon ( QMessageBox::Warning );
+            qmASIOWarningBox.exec();
 
-            if ( QMessageBox::Yes == QMessageBox::information ( nullptr, APP_NAME, sErrorMessage, QMessageBox::Yes | QMessageBox::No ) )
+            if ( qmASIOWarningBox.clickedButton() == btnASIOSettings )
             {
-                LoadAndInitializeFirstValidDriver ( true );
+                vsErrorList = LoadAndInitializeFirstValidDriver ( true );
+            }
+
+            if ( vsErrorList.isEmpty() ) {
+                return strReturn;
             }
```

Comment:
As mentioned before, I think this is not enough.

Reply:
It looks like it's what's needed and removes my concern at https://github.com/jamulussoftware/jamulus/pull/2168/files#r770944946